protostar/commands/test/cheatcodes/deploy_cheatcode.py

```diff
@@ -27,5 +32,26 @@ def deploy_prepared(
         self,
         prepared: PreparedContract,
     ):
-        # TODO(mkaput): Reimplement this.
-        raise NotImplementedError("Deploy cheatcode is not ported to Cairo 10 yet.")
+        deploy_tx_hash = calculate_deploy_transaction_hash(
+            version=0,
+            contract_address=prepared.contract_address,
+            constructor_calldata=prepared.constructor_calldata,
+            chain_id=self.general_config.chain_id.value,
+        )
+
+        tx = InternalDeploy(
+            contract_address=prepared.contract_address,
+            contract_address_salt=prepared.salt,
+            contract_hash=to_bytes(prepared.class_hash),
+            constructor_calldata=prepared.constructor_calldata,
+            hash_value=deploy_tx_hash,
+            version=0,
+        )
+
+        asyncio.run(self._run_deploy_tx(tx))
```

Comment:
There's a method called `sync_apply_state_updates` in `InternalStateTransaction` which I think we can use here instead of calling into asyncio

Reply:
OK i will check that one out

Reply:
this one is a bummer because the `cheatable_state` does not pass instance check of a sync state, which crashes in case i use `sync_apply_state_updates`.

Reply:
I think you can call copy and apply manually in the same way copy_and_apply does internally. I have asked StarkWare people about this internally.